homeassistant/components/shelly/config_flow.py

```diff
@@ -39,13 +40,13 @@ async def validate_input(
     Data has the keys from DATA_SCHEMA with values provided by the user.
     """
 
-    options = aioshelly.ConnectionOptions(
+    options = aioshelly.common.ConnectionOptions(
         host, data.get(CONF_USERNAME), data.get(CONF_PASSWORD)
     )
     coap_context = await get_coap_context(hass)
 
     async with async_timeout.timeout(AIOSHELLY_DEVICE_TIMEOUT_SEC):
-        device = await aioshelly.Device.create(
+        device = await BlockDevice.create(
```

Comment:
Do we handle if the generation is not supported?

Reply:
I guess that will come in a future PR.

Reply:
Yes